src/app/delegates/actions.ts

```diff
@@ -105,3 +106,17 @@ export async function fetchAllDelegatorsInChainsForAddress(
 ) {
   return getAllDelegatorsInChainsForAddress({ addressOrENSName });
 }
+
+export async function balanceOf(
+  address: string
+) {
+  return OptimismContracts.token.contract.balanceOf(address);
+}
+
+export const fetchConnectedDelegate = async (address: string) => {
+  return await Promise.all([
+    fetchDelegate(address),
+    fetchAllDelegatorsInChainsForAddress(address),
+    balanceOf(address)
+  ]);
+};
```

Comment:
I think in order to for the browser to parallelize the requests, all relevant functions should be merged into one

Reply:
my miss, this is correct